File: C#/envio.cs
using System;
using System.IO;
using System.Net;
using System.Text;

class Program
{
    static void Main()
    {
        // URL do endpoint da API
        var url = "https://api.mailgrid.net.br/send/";

        try
        {
            // Criação da requisição HTTP POST
            var httpRequest = (HttpWebRequest)WebRequest.Create(url);
            httpRequest.Method = "POST";

            // Definindo os cabeçalhos da requisição
            httpRequest.Headers["Authorization"] = "Bearer SEU_TOKEN"; // Corrigido para Bearer Token (se necessário)
            httpRequest.ContentType = "application/json"; // Especificando o tipo do conteúdo

            // Dados da requisição, que serão enviados no corpo da requisição
            var data = @"{
                            ""host_smtp"": ""HOST-SMTP"",
                            ""usuario_smtp"": ""USUARIO-SMTP"",
                            ""senha_smtp"": ""SENHA-SMTP"",
                            ""emailRemetente"": ""EMAIL-REMETENTE"",
                            ""nomeRemetente"": ""NOME-REMETENTE"",
                            ""emailDestino"": [""postmaster@mailgrid.com.br"", ""dev@mailgrid.com.br""],
                            ""assunto"": ""Teste de envio via API JSON"",
                            ""mensagem"": ""mensagem de teste da API JSON""
                        }";

            // Escrevendo os dados no corpo da requisição
            using (var streamWriter = new StreamWriter(httpRequest.GetRequestStream(), Encoding.UTF8))
            {
                streamWriter.Write(data);
            }

            // Obtenção da resposta HTTP
            using (var httpResponse = (HttpWebResponse)httpRequest.GetResponse())
            {
                // Lê o conteúdo da resposta
                using (var streamReader = new StreamReader(httpResponse.GetResponseStream()))
                {
                    var result = streamReader.ReadToEnd();
                    Console.WriteLine($"Resposta da API: {result}");
                }

                // Exibe o código de status da resposta HTTP
                Console.WriteLine($"Código de status: {httpResponse.StatusCode}");
            }
        }
        catch (WebException webEx)
        {
            // Captura falhas de requisição, como erros de rede, URL incorreta ou erro no servidor
            using (var streamReader = new StreamReader(webEx.Response.GetResponseStream()))
            {
                var errorResponse = streamReader.ReadToEnd();
                Console.WriteLine($"Erro na requisição: {errorResponse}");
            }
        }
        catch (Exception ex)
        {
            // Captura outros tipos de exceções
            Console.WriteLine($"Erro inesperado: {ex.Message}");
        }
    }
}

File: php/envio.php
// Inicializa a sessão cURL
$curl = curl_init();

// Define as opções cURL usando um array associativo
curl_setopt_array($curl, array(
 // URL da API mailgrid para envio de email
 CURLOPT_URL =>'https://api.mailgrid.net.br/sendmail/',
	
 // Faz com que o cURL retorne a resposta como string
 CURLOPT_RETURNTRANSFER =>true,

 // Define a codificação de resposta para ser aceita
 CURLOPT_ENCODING =>'',

 // Define o número máximo de redirecionamentos
 CURLOPT_MAXREDIRS =>10,

 // Define o tempo máximo de espera da requisição
 CURLOPT_TIMEOUT =>0,

 // Permite o cURL seguir redirecionamentos automaticamente
 CURLOPT_FOLLOWLOCATION =>true,

 // Define a versão do protocolo HTTP a ser usado
 CURLOPT_HTTP_VERSION =>CURL_HTTP_VERSION_1_1,

 // Define o método da requisição como POST
 CURLOPT_CUSTOMREQUEST =>'POST',

 // Define os dados do corpo da requisição em formato JSON
 CURLOPT_POSTFIELDS => '{
    "host_smtp": "HOST-SMTP", // Substitua por seu host SMTP
    "usuario_smtp": "USUARIO-SMTP", // Substitua pelo usuário SMTP
    "senha_smtp": "SENHA-SMTP", // Substitua pela senha SMTP
    "emailRemetente": "EMAIL-REMETENTE", // Email do remetente
    "nomeRemetente": "NOME-REMETENTE", // Nome do remetente
    "emailDestino": ["destinatario1@dominio.com.br", "destinatario2@dominio.com.br"], // Emails dos destinatários
    "assunto": "Teste de envio com anexo via API JSON", // Assunto do email
    "mensagemAnexos": [
	    {
	       "name": "anexoexemplo.pdf", // Nome do arquivo
	       "type": "application/pdf", // Tipo MIME do arquivo
	        "content": "UEAEAAAAAAAAAAAAAAAAAAAAA/..." // Conteúdo do arquivo codificado em base64
	    }
    ],
    "mensagem": "Mensagem de teste da API com anexo. Atenção, o arquivo codificado acima não funciona, é um exemplo."
    "mensagemTipo": "html", // Tipo de mensagem (HTML)
    "mensagemEncoding": "base64", // Codificação do arquivo anexo (base64)
    "mensagemAlt": "Mensagem de teste da API com anexo. Atenção, o arquivo codificado acima não funciona, é um exemplo."
}',

	// Define o cabeçalho da requisição, informando que o corpo está em JSON
	CURLOPT_HTTPHEADER => array(
		'Content-Type: application/json'
	),
));

// Executa a requisição cURL e armazena a resposta
$response = curl_exec($curl);

// Fecha a sessão cURL
curl_close($curl);

// Exibe a resposta recebida da API
echo $response;
